### ISSUE TITLE
No artifacts were published

### ISSUE THREAD

**Issue by isage** [appveyor/ci#3597]
Here's build log https://ci.appveyor.com/project/nxengine/nxengine-evo/builds/36908628/job/11tc3v72i7juq3vx#L9642
It definitely states that artifact was published, yet deployment states "No artifacts were published"
Here's my https://github.com/nxengine/nxengine-evo/blob/master/appveyor.yml

I've tried:
- quoting path in artifacts.
- separate deploy sections with regex match for artifact
- separate deploy sections with name of artifact
- separate deploy sections with filename of artifact
- escaping asterisk in filters/artifacts

Windows builder successfully deploys zip, but osx/linux builders can't find dmg/AppImage artifacts even though two lines above they were successfully pushed (and I can see them on artifacts tab)

**Comment by FeodorFitsner** [comment]
Right now you are using an overridden `appveyor.yml` defined in gist, so https://github.com/nxengine/nxengine-evo/blob/master/appveyor.yml is not currently used. That build you gave a link to is deleted already.

**Comment by isage** [comment]
Yeah, sorry, i was experimenting. Current build with appveyor.yml from repo: https://ci.appveyor.com/project/nxengine/nxengine-evo/builds/36929842

**Comment by FeodorFitsner** [comment]
Will be looking into that.

**Comment by FeodorFitsner** [comment]
Ah, seems like I've found the issue! You are pushing artifact on `before_deploy` step, but artifacts must be "ready" by the deployment start. Move [all 3 lines](https://github.com/nxengine/nxengine-evo/blob/47c8161dd9accdd5884d1040dce908559f026f2d/appveyor.yml#L98-L100) from `before_deploy` section to `build_script` and it should do the trick.

I'm sorry you've spent so much time fighting this!

**Comment by isage** [comment]
I don't really want to move it to build script, because that will increase build time of regular commits.
And pushing from `before_deploy` works fine for windows build

**Comment by isage** [comment]
Well, guess I'll just check for APPVEYOR_REPO_TAG in `after_build`, but that seems strange anyway (especially windows version working, while others - not). I'll leave this case open, but feel free to close, if you think everything is as it should be

**Comment by FeodorFitsner** [comment]
There is a different branch/implementation of build agent on Mac and Linux. Sure, the behaviour must be similar across all platforms - let's keep the issue open.
